integration-tests/source/IntegrationTestResourceHandler.cpp

```diff
@@ -323,3 +329,102 @@ std::string IntegrationTestResourceHandler::GetResourceId(const std::string &res
 {
     return resource.substr(resource.find('/'));
 }
+vector<ActiveViolation> IntegrationTestResourceHandler::GetViolations(const string &thingName)
+{
+    vector<ActiveViolation> violations;
+
+    ListActiveViolationsRequest request;
+    request.SetThingName(thingName);
+
+    ListActiveViolationsOutcome outcome = iotClient.ListActiveViolations(request);
+
+    if (!outcome.IsSuccess())
+    {
+        printf(
+            "Failed to List Active Violations for: %s\n%s\n",
+            thingName.c_str(),
+            outcome.GetError().GetMessage().c_str());
+    }
+    else
+    {
+        violations = outcome.GetResult().GetActiveViolations();
+    }
+    return violations;
+}
+void IntegrationTestResourceHandler::CreateAndAttachSecurityProfile(
+    const string &profileName,
+    const vector<std::string> &metrics)
+{
+    MetricValue metricValue;
+    metricValue.SetCount(1L);
+
+    BehaviorCriteria criteria;
+    criteria.SetComparisonOperator(ComparisonOperator::less_than);
+    criteria.SetDurationSeconds(300);
+    criteria.SetValue(metricValue);
+    criteria.SetConsecutiveDatapointsToAlarm(1);
+    criteria.SetConsecutiveDatapointsToClear(1);
+
+    CreateSecurityProfileRequest request;
+    request.SetSecurityProfileName(profileName);
+
+    for (const string &metric : metrics)
+    {
+        Behavior behavior;
+        behavior.SetCriteria(criteria);
+        behavior.SetMetric(metric);
+        behavior.SetName(metric);
+
+        request.AddBehaviors(behavior);
+    }
+
+    CreateSecurityProfileOutcome outcome = iotClient.CreateSecurityProfile(request);
+
+    if (!outcome.IsSuccess())
+    {
+        printf(
+            "Failed to create Security Profile: %s\n%s\n",
+            profileName.c_str(),
+            outcome.GetError().GetMessage().c_str());
+    }
+
+    AttachSecurityProfile(profileName);
+}
+
+void IntegrationTestResourceHandler::AttachSecurityProfile(const std::string &profileName)
```

Comment:
Curious to know whether we will ever call this method on its own (not as a part of CreateAndAttachSecurityProfile). I think we either make this method private or remove the call from line 391 and call this directly from test setup.

Reply:
I think I meant to make it private but made it protected by mistake. Will change.